decidim-initiatives/app/forms/decidim/initiatives/initiative_form.rb

```diff
@@ -42,6 +45,26 @@ def scope_exists
 
         errors.add(:scope_id, :invalid) unless InitiativesTypeScope.where(decidim_initiatives_types_id: type_id, decidim_scopes_id: scope_id).exists?
       end
+
+      # This method will add an error to the `attachment` field only if there's
+      # any error in any other field. This is needed because when the form has
+      # an error, the attachment is lost, so we need a way to inform the user of
+      # this problem.
+      def notify_missing_attachment_if_errored
+        return if attachment.blank?
+
+        errors.add(:attachment, :needs_to_be_reattached) if errors.any?
+      end
+
+      def trigger_attachment_errors
+        return if attachment.blank?
+
+        return if attachment.valid?
+
+        attachment.errors.each do |error|
+          errors.add(:attachment, error)
+        end
+      end
```

Comment:
This little trick solves a part of the problem. Due to wicked's wizard, the validation on the form is not done until the final step, so we ensure that validation are triggered before.